### ISSUE TITLE
Unsupported Browser

### ISSUE THREAD

**Issue by gianpelisser** [nirewen/discord-netflix#79]
Error code: M7121-3078
Unsupported or non-compliant browser.

**Comment by Azure-Fang** [comment]
Add another name to that list.

Launch and login work fine, boots straight to the home screen, but after selecting something to watch, it attempts to load the selection followed by the following message:

> Whoops, something went wrong...
>
> **Browser Not Supported**
>
> Your browser is no longer supported. Please upgrade or try another browser.
>
> For additional assistance resolving this issue, please visit out Help Center.
> https://help.netflix.com/support/109143
>
> Error Code: M7121-3078

**Comment by scoo12** [comment]
I'm having the same problem since yesterday

**Comment by GlobalShock** [comment]
I also have this issues.

Browser Not Supported

Your browser is no longer supported. Please upgrade or try another browser.

For additional assistance resolving this issue, please visit out Help Center.
https://help.netflix.com/support/109143

**Comment by tautod15** [comment]
me too

**Comment by Niguella** [comment]
Add myself to the list.

**Comment by EGXZ** [comment]
Yes. I've been warned couple times that 16 may the browser I were using will not be supported anymore. And that's the day I guess.

**Comment by alterlove** [comment]
Not working anymore

**Comment by SirNoble** [comment]
Same as everyone else, "unsupported browser"

**Comment by eliac7** [comment]
Add me too here, same issue!

**Comment by nirewen** [comment]
Recently I discovered that this error is due to the outdated version of electron-widevinecdm I'm using, which is 5.0.2, that used to work on all systems. If I update to v7.0.0, the latest version, it works again, but not for Windows. So the reason I'm holding this for so long because is because it's very hard to just drop Windows support and that's it. I don't have a Widevine certificate and most likely won't get one. I can try another browser other than Chromium on Electron, but even with that, I still need a Widevine component. And if I were to switch to another browser, it would take months to study and implement, and the language might not be a language I know. So either I drop Windows support for good or until I find another way to add the Widevine component, or I archive the repo for good. Either way is a death to one of the sides...

**Comment by GlobalShock** [comment]
> Recently I discovered that this error is due to the outdated version of electron-widevinecdm I'm using, which is 5.0.2, that used to work on all systems. If I update to v7.0.0, the latest version, it works again, but not for Windows. So the reason I'm holding this for so long because is because it's very hard to just drop Windows support and that's it. I don't have a Widevine certificate and most likely won't get one. I can try another browser other than Chromium on Electron, but even with that, I still need a Widevine component. And if I were to switch to another browser, it would take months to study and implement, and the language might not be a language I know. So either I drop Windows support for good or until I find another way to add the Widevine component, or I archive the repo for good. Either way is a death to one of the sides...

So buddy there is no solution to fix this issue ?

**Comment by nirewen** [comment]
> > Recently I discovered that this error is due to the outdated version of electron-widevinecdm I'm using, which is 5.0.2, that used to work on all systems. If I update to v7.0.0, the latest version, it works again, but not for Windows. So the reason I'm holding this for so long because is because it's very hard to just drop Windows support and that's it. I don't have a Widevine certificate and most likely won't get one. I can try another browser other than Chromium on Electron, but even with that, I still need a Widevine component. And if I were to switch to another browser, it would take months to study and implement, and the language might not be a language I know. So either I drop Windows support for good or until I find another way to add the Widevine component, or I archive the repo for good. Either way is a death to one of the sides...
> 
> So buddy there is no solution to fix this issue ?

Yeah, not for Windows... If I update the version and release a new Windows installer, it will not work, insteadm will say it's missing the widevine component

**Comment by EGXZ** [comment]
That's really sad buddy, honestly. I felt so sad for you. And alll of us.

You put so much work and love on this project and seen it being archived it's sad.

The solution is in your hands, you just have to choose what you decide to do :')

Love you.

**Comment by tautod15** [comment]
Is there an "unofficial" work around for this?

**Comment by SerenModz21** [comment]
im having the same issue also, we are getting this because the wrapper needs updating to the latest version of netflix

**Comment by gianpelisser** [comment]
> > > Recently I discovered that this error is due to the outdated version of electron-widevinecdm I'm using, which is 5.0.2, that used to work on all systems. If I update to v7.0.0, the latest version, it works again, but not for Windows. So the reason I'm holding this for so long because is because it's very hard to just drop Windows support and that's it. I don't have a Widevine certificate and most likely won't get one. I can try another browser other than Chromium on Electron, but even with that, I still need a Widevine component. And if I were to switch to another browser, it would take months to study and implement, and the language might not be a language I know. So either I drop Windows support for good or until I find another way to add the Widevine component, or I archive the repo for good. Either way is a death to one of the sides...
> > 
> > 
> > So buddy there is no solution to fix this issue ?
> 
> Yeah, not for Windows... If I update the version and release a new Windows installer, it will not work, insteadm will say it's missing the widevine component

And if you create a new only for windows, and let this one for linux and Mac?

**Comment by OsqaiHD** [comment]
Please update <3

**Comment by lamjed001** [comment]
We need update ♥ @nirewen @Wist9063 @NovusTheory

**Comment by enlyth** [comment]
For those of you who want to get this running manually on win-x64:

Go to the following path:
`<repo>\node_modules\electron-widevinecdm\widevine\win32_x64\_platform_specific\win_x64`

Now use a tool like [Everything](https://www.voidtools.com/) to find all instances of `widevinecdm.dll` on your computer, there will be quite a few.

Replace `widevinecdm.dll` and `widevinecdm.dll.sig` in the node_modules directory with a newer one from your computer, I tried the Chrome v74 one first, gave a different error, and then one from my Steam directory worked fine (`C:\Users\enlyth\AppData\Local\Steam\widevine\win-x64`).

Don't `npm install` as it will re-write the file again.

**Comment by UntamedBananas** [comment]
> For those of you who want to get this running manually on win-x64:
> 
> Go to the following path:
> `<repo>\node_modules\electron-widevinecdm\widevine\win32_x64\_platform_specific\win_x64`
> 
> Now use a tool like [Everything](https://www.voidtools.com/) to find all instances of `widevinecdm.dll` on your computer, there will be quite a few.
> 
> Replace `widevinecdm.dll` and `widevinecdm.dll.sig` in the node_modules directory with a newer one from your computer, I tried the Chrome v74 one first, gave a different error, and then one from my Steam directory worked fine (`C:\Users\enlyth\AppData\Local\Steam\widevine\win-x64`).
> 
> Don't `npm install` as it will re-write the file again.

Is it possible if you could give a fresher fish like myself a brief/quick rundown on how to properly replace the module to the different directory you mentioned above?

**Comment by enlyth** [comment]
> Is it possible if you could give a fresher fish like myself a brief/quick rundown on how to properly replace the module to the different directory you mentioned above?

Just literally replace the dll and dll.sig files which are inside of the cloned discord-netflix repo you have on your drive with the one from your Steam folder for example. If you'd like some help add me on discord enlyth#1177

**Comment by UntamedBananas** [comment]
@enlyth Added you on Discord. Thank you for the help man.

**Comment by ChuckNoxis** [comment]
> Replace `widevinecdm.dll` and `widevinecdm.dll.sig` in the node_modules directory with a newer one (from the Steam directory worked fine).

Worked perfectly for me

Thanks @enlyth ! :)

**Comment by GlobalShock** [comment]
Works like charm ! Thank you for your help Enlyth.

**Comment by dpvdberg** [comment]
Thanks for the fix @enlyth

**Comment by OsqaiHD** [comment]
Please update @nirewen ?

**Comment by virgovelour** [comment]
It's telling me that the browser is unsupported...

**Comment by sollazzo08** [comment]
So I am on mac and running into the same issue

**Comment by Phnessed** [comment]
> > Is it possible if you could give a fresher fish like myself a brief/quick rundown on how to properly replace the module to the different directory you mentioned above?
> 
> Just literally replace the dll and dll.sig files which are inside of the cloned discord-netflix repo you have on your drive with the one from your Steam folder for example. If you'd like some help add me on discord enlyth#1177

I sent you a request on discord, my tag is: lty.Wolf+18-cwᵐᵗ#4258, would like some help with this, newbie here lol!

**Comment by alterlove** [comment]
Add on discord if you need help fixing it on Windows. Lucas Lallemant#0023

Enviado desde mi iPhone

**Comment by gianpelisser** [comment]
HOW TO INSTALL in [ENGLISH] and Portuguese
COMO INSTALAR em [Português] e Inglês

[ENGLISH] - HOW TO INSTALL/FIX
First install the node-v7.0.0-x64 (I think a newer version will work the same).
Then install Netflix

Now you need to configure as it will give incompatible browser error.
Go to %appdata%
And you need to copy the files inside that folder
AppData\Local\Steam\widevine\win-x64
and copy them into
AppData\Local\Programs\Netflix\resources\app.asar.unpacked\node_modules\electron-widevinecdm\widevine\win32_x64_platform_specific\win_x64
Done

Special thanks to Lucas Lallemant and Enlyth.

============[Portuguese - Brazilian]============

[Português] - Como Instalar / Arrumar

============[Instalar]============

Primeiro instale o node-v7.0.0-x64 (Talvez possa funcionar com uma versão mais recente).
Depois instale o Netflix-WebSetup-2.2.1

============[Configurando/Arrumando - problema de Navegador incompatível]============
Entre na pasta APP DATA / LOCAL
Abra uma pasta e escreva %appdata%

Volte 1 pasta e entre na pasta Local e depois procure pela pasta Steam:
AppData\Local\Steam\widevine\win-x64

Copie os arquivos dentro da pasta win-x64
Se não tiver essa pasta, você precisa instalar a STEAM

agora entra na pasta [Local/Programs/Netflix]
AppData\Local\Programs\Netflix\resources\app.asar.unpacked\node_modules\electron-widevinecdm\widevine\win32_x64_platform_specific\win_x64

Cole os arquivos da pasta steam/win-x64 dentro da pasta win32_x64_platform_specific/win_x64

Pronto!

Um Obrigado Especial para Lucas Lallemant e Enlyth.
[How to install - Como instalar.txt](https://github.com/nirewen/discord-netflix/files/3395505/How.to.install.-.Como.instalar.txt)

**Comment by alterlove** [comment]
But you dont need node to fix it, just copying steam files will make it work.

Enviado desde mi iPhone

**Comment by Lite3000** [comment]
> [ENGLISH] - HOW TO INSTALL/FIX
> First install the node-v7.0.0-x64 (I think a newer version will work the same).
> Then install Netflix
> 
> Now you need to configure as it will give incompatible browser error.
> Go to %appdata%
> And you need to copy the files inside that folder
> AppData\Local\Steam\widevine\win-x64
> and copy them into
> AppData\Local\Programs\Netflix\resources\app.asar.unpacked\node_modules\electron-widevinecdm\widevine\win32_x64_platform_specific\win_x64

When I do this, the error I get is "Missing Component". It says that it's missing WideVineCDM

**Comment by alterlove** [comment]
**This method stopped to work, so this is officially dead.**

**Comment by HQv2** [comment]
I'm getting this error again. [This fix](https://github.com/nirewen/discord-netflix/issues/79#issuecomment-500622877) is no longer working.
